src/mbgl/renderer/layers/render_custom_layer.cpp

```diff
@@ -67,7 +69,7 @@ void RenderCustomLayer::render(PaintParameters& paintParameters) {
         host = impl(baseImpl).host;
         MBGL_CHECK_ERROR(host->initialize());
     }
-
+#ifdef MBGL_RENDER_BACKEND_OPENGL
```

Comment:
Only code 

```
    auto& glContext = static_cast<gl::Context&>(paintParameters.context);

    // Reset GL state to a known state so the CustomLayer always has a clean slate.
    glContext.bindVertexArray = 0;
    glContext.setDepthMode(paintParameters.depthModeForSublayer(0, gfx::DepthMaskType::ReadOnly));
    glContext.setStencilMode(gfx::StencilMode::disabled());
    glContext.setColorMode(paintParameters.colorModeForRenderPass());
    glContext.setCullFaceMode(gfx::CullFaceMode::disabled());
```

and 

```
    // Reset the view back to our original one, just in case the CustomLayer changed
    // the viewport or Framebuffer.
    paintParameters.backend.getDefaultRenderable().getResource<gl::RenderableResource>().bind();
    glContext.setDirtyState();
```

is OpenGL specific - the rest (L.75 and [L.84-L.99]) shouldn't be wrapped by #ifdef MBGL_RENDER_BACKEND_OPENGL.  We need to move it under platform agnostic approach. That could be done in follow up patches.

Reply:
In addition, under #ifdef we need to have runtime wrap (gfx::Backend::GetType() == gfx::Backend::Type::OpenGL)

Reply:
done